### PR TITLE
Remove deal damage data

### PR DESCRIPTION
**Pull request by Paper-2**
# Description

This PR removes `DealDamageData` from the code base

## Related issue(s)

#44

## Additional notes

`EffectDamageAll` is the only exception due to being removed in a future PR.

### CODE REVIEW

File: Entity/Components/HealthComponent.gd [hunk 1]
```diff
@@ -17,16 +17,16 @@ func _ready() -> void:
 # If caster is null, we assume that the damage came from an unknown source,
 # so status won't calculate.
 # Use a negative damage value if you want to heal.
-func deal_damage(damage_data: DealDamageData) -> void:
-	var damage: float = damage_data.damage
+func deal_damage(incoming_damage, by_caster) -> void:
+	var damage: float = incoming_damage
```
Comment (Turtyo) [inline Entity/Components/HealthComponent.gd:21]: i don't think you need to create a new variable for that

File: Entity/Components/HealthComponent.gd [hunk 2]
```diff
@@ -17,16 +17,16 @@ func _ready() -> void:
 # If caster is null, we assume that the damage came from an unknown source,
 # so status won't calculate.
 # Use a negative damage value if you want to heal.
-func deal_damage(damage_data: DealDamageData) -> void:
-	var damage: float = damage_data.damage
+func deal_damage(incoming_damage, by_caster) -> void:
```
Comment (Turtyo) [inline Entity/Components/HealthComponent.gd:20]: missing typing

File: Status/Debuffs/Debuff_Poison.gd [hunk 3]
```diff
@@ -18,8 +18,8 @@ func init_status(in_caster: Entity, in_target: Entity) -> void:
 
 # @Override
 func on_turn_start() -> void:
-	var damage_data: DealDamageData = DealDamageData.new()
-	damage_data.damage = status_power
-	damage_data.caster = status_caster
+
+	var damage = status_power
```
Comment (Turtyo) [inline Status/Debuffs/Debuff_Poison.gd:22]: missing typing

**Review by Turtyo — CHANGES_REQUESTED** [review]
You forgot the typing everywhere

File: Entity/Components/HealthComponent.gd (hunk 1 above)
Comment (Paper-2) [reply Entity/Components/HealthComponent.gd:21]: True.

File: Status/Debuffs/Debuff_Poison.gd [hunk 4]
```diff
@@ -18,8 +18,8 @@ func init_status(in_caster: Entity, in_target: Entity) -> void:
 
 # @Override
 func on_turn_start() -> void:
-	var damage_data: DealDamageData = DealDamageData.new()
-	damage_data.damage = status_power
-	damage_data.caster = status_caster
+
+	var damage: float  = status_power
+	var caster: Entity = status_caster
```
Comment (Tysterman74) [inline Status/Debuffs/Debuff_Poison.gd:23]: I don't think we need these local variables here, and can instead just use `status_power` and `status_caster` instead.

File: Tests/test_health.gd [hunk 5]
```diff
@@ -52,48 +52,48 @@ func after_each():
 
 
 func test_take_damage():
-	var deal_damage_data = DealDamageData.new()
-	deal_damage_data.damage = 5.0
-	deal_damage_data.caster = _player
+
```
Comment (Tysterman74) [inline Tests/test_health.gd:55]: Let's remove the extra whitespace here and in the other test functions.

**Review by Tysterman74 — CHANGES_REQUESTED** [review]

File: Status/Debuffs/Debuff_Poison.gd (hunk 4 above)
Comment (Turtyo) [reply Status/Debuffs/Debuff_Poison.gd:23]: I agree, can just directly use the variables

**Review by Turtyo — DISMISSED** [review]

**Review by Turtyo — APPROVED** [review]

**Review by Tysterman74 — APPROVED** [review]

**Review by cheesefrycook — APPROVED** [review]

**Pull request merged**